uniffi/src/ffi/foreigncallbacks.rs

```diff
@@ -131,8 +132,12 @@ use std::sync::atomic::{AtomicUsize, Ordering};
 ///   passing individual arguments to the user's callback.
 /// * `buf_ptr` is a pointer to where the resulting buffer will be written. UniFFI will allocate a
 ///   buffer to write the result into.
-/// * A callback returns `-1` if an unexpected error happens, `0` if the call succeeded without a
-///   return value or `1` if data was written into the output buffer.
+/// * A callback returns:
+///    - `-2` An error occured that was serialized to buf_ptr
+///    - `-1` An unexpected error ocurred
+///    - `0` is a deprecated way to signal that if the call succeeded, but did not modify buf_ptr
```

Comment:
So I *think* this means that even a function returning `Ok(())` will modify `buf_ptr`, so this is deprecated because it can only happen for a function which returns a void not in a Result - which is deprecated. If that's correct it might be worth expanding this (or not - I'm really just trying to understand how this is "deprecated" from the POV of the user)